src/content/blog/year-in-review-2024.mdx

```diff
@@ -0,0 +1,716 @@
+
+```astro title="src/pages/index.astro"
+---
+import Avatar from "../components/Avatar.astro";
+---
+
+<!-- Other content here -->
+
+<Avatar server:defer />
+```
+
+And that's it! The page is statically generated, while the `Avatar` component is server-rendered on demand. In addition, you can also specify fallback content to display while the server island is being loaded, akin to React's [`Suspense`](https://react.dev/reference/react/Suspense):
+
+```astro title="src/pages/index.astro"
+---
+import Avatar from "../components/Avatar.astro";
+import GenericAvatar from "../components/GenericAvatar.astro";
+---
+
+<Avatar server:defer>
+  <GenericAvatar slot="fallback" />
+</Avatar>
+```
+
+If you'd like a deeper explanation of how Server Islands differ from Astro's standard Prerendering or On-demand Rendering, check out [my article (in Japanese)](https://zenn.dev/morinokami/articles/astro-server-islands-vs-nextjs-ppr). It compares Next.js's rendering methods and Astro's Server Islands, giving you a broader understanding of modern rendering techniques.
+
+[Check out the official docs to learn more about Server Islands](https://docs.astro.build/en/guides/server-islands/).
+
+### Astro DB
+
+In March 2024, Astro introduced the [`@astrojs/db`](https://docs.astro.build/en/guides/integrations-guide/db/) package for database management. At first release, it was intended that production environments would connect to a managed database service called Astro Studio. As of now, you can connect to any [libSQL](https://github.com/tursodatabase/libsql) server. For more about Astro Studio, see [Related Projects](#related-projects).
+
+To use Astro DB, you first need to define your table schema in `db/config.ts` file:
+
+```js title="db/config.ts"
+import { defineTable, column } from "astro:db";
+
+const Author = defineTable({
+  columns: {
+    id: column.number({ primaryKey: true }),
+    name: column.text(),
+  }
+});
+
+const Comment = defineTable({
+  columns: {
+    authorId: column.number({ references: () => Author.columns.id }),
+    body: column.text(),
+  }
+});
+
+export default defineDb({
+  tables: { Author, Comment },
+});
+```
+
+Then, in `db/seed.ts`, export a function that seeds your database:
+
+```js title="db/seed.ts"
+import { db, Comment, Author } from "astro:db";
+
+export default async function() {
+  await db.insert(Author).values([
+    { id: 1, name: "Kasim" },
+    { id: 2, name: "Mina" },
+  ]);
+
+  await db.insert(Comment).values([
+    { authorId: 1, body: "Hope you like Astro DB!" },
+    { authorId: 2, body: "Enjoy!"},
+  ])
+}
+```
+
+When you spin up the development server, the database is automatically created and seeded.
+
+Querying the database go through the [`Drizzle`](https://orm.drizzle.team/) client. The `db` client imported from `astro:db` is automatically configured to connect to the correct database, so you can just start querying right away:
```

Comment:
Another nit, but I don’t think we need the code backticks around “Drizzle” as it’s just the project name not a specific code API:

```suggestion
Querying the database go through the [Drizzle](https://orm.drizzle.team/) client. The `db` client imported from `astro:db` is automatically configured to connect to the correct database, so you can just start querying right away:
```